sycl/source/detail/device_info.hpp

```diff
@@ -2210,6 +2210,191 @@ inline sycl::device get_device_info_host<
                       PI_ERROR_INVALID_DEVICE);
 }
 
+template <>
+inline std::vector<ext::oneapi::experimental::forward_progress_guarantee>
+get_device_info_host<
+    ext::oneapi::experimental::info::device::work_group_progress_capabilities<
+        ext::oneapi::experimental::execution_scope::root_group>>() {
+  return {
+      ext::oneapi::experimental::forward_progress_guarantee::weakly_parallel};
+}
+
+template <>
+inline std::vector<ext::oneapi::experimental::forward_progress_guarantee>
+get_device_info_host<
+    ext::oneapi::experimental::info::device::sub_group_progress_capabilities<
+        ext::oneapi::experimental::execution_scope::root_group>>() {
+  return {
+      ext::oneapi::experimental::forward_progress_guarantee::weakly_parallel};
+}
+
+template <>
+inline std::vector<ext::oneapi::experimental::forward_progress_guarantee>
+get_device_info_host<
+    ext::oneapi::experimental::info::device::sub_group_progress_capabilities<
+        ext::oneapi::experimental::execution_scope::work_group>>() {
+  return {
+      ext::oneapi::experimental::forward_progress_guarantee::weakly_parallel};
+}
+
+template <>
+inline std::vector<ext::oneapi::experimental::forward_progress_guarantee>
+get_device_info_host<
+    ext::oneapi::experimental::info::device::work_item_progress_capabilities<
+        ext::oneapi::experimental::execution_scope::root_group>>() {
+  return {
+      ext::oneapi::experimental::forward_progress_guarantee::weakly_parallel};
+}
+
+template <>
+inline std::vector<ext::oneapi::experimental::forward_progress_guarantee>
+get_device_info_host<
+    ext::oneapi::experimental::info::device::work_item_progress_capabilities<
+        ext::oneapi::experimental::execution_scope::work_group>>() {
+  return {
+      ext::oneapi::experimental::forward_progress_guarantee::weakly_parallel};
+}
+
+template <>
+inline std::vector<ext::oneapi::experimental::forward_progress_guarantee>
+get_device_info_host<
+    ext::oneapi::experimental::info::device::work_item_progress_capabilities<
+        ext::oneapi::experimental::execution_scope::sub_group>>() {
+  return {
+      ext::oneapi::experimental::forward_progress_guarantee::weakly_parallel};
+}
+
+const int forward_progress_guarantee_size = 3;
+
+template <typename ReturnT>
+struct get_device_info_impl<
+    ReturnT,
+    ext::oneapi::experimental::info::device::work_group_progress_capabilities<
+        ext::oneapi::experimental::execution_scope::root_group>> {
+  static ReturnT get(const DeviceImplPtr &Dev) {
+    using forward_progress_guarantee =
+        ext::oneapi::experimental::forward_progress_guarantee;
+    using execution_scope = ext::oneapi::experimental::execution_scope;
+    ReturnT res;
```

Comment:
The file is a little inconsistent with the coding style, but unless the file has a consistent coding style, we should use the [LLVM Coding Standard](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly) mandated naming. This does not apply to public interfaces.